kornia/geometry/bbox_v2.py

```diff
@@ -0,0 +1,585 @@
+
+        boxes = boxes if self._is_batch else boxes.squeeze(0)
+        return boxes
+
+    def to_mask(self, height: int, width: int) -> torch.Tensor:
+        """Convert 2D bounding boxes to masks. Covered area is 1 and the remaining is 0.
+
+        Args:
+            height: height of the masked image/images.
+            width: width of the masked image/images.
+
+        Returns:
+            the output mask tensor, shape of :math:`(N, width, height)` or :math:`(B,N, width, height)` and dtype of
+            boxes.
+
+        Note:
+            It is currently non-differentiable.
+
+        Examples:
+            >>> boxes_xyxy = torch.tensor([[
+            ...        [1., 1.],
+            ...        [4., 1.],
+            ...        [4., 3.],
+            ...        [1., 3.],
+            ...   ]])  # 1x4x2
+            >>> boxes = Boxes.from_tensor(boxes)
+            >>> boxes.to_mask(5, 5)
+            tensor([[[0., 0., 0., 0., 0.],
+                     [0., 1., 1., 1., 0.],
+                     [0., 1., 1., 1., 0.],
+                     [0., 0., 0., 0., 0.],
+                     [0., 0., 0., 0., 0.]]])
+        """
+        if self._is_batch:  # (B, N, 4, 2)
+            mask = torch.zeros(
+                (self._boxes.shape[0], self._boxes.shape[1], height, width), dtype=self.dtype, device=self.device
+            )
+        else:  # (N, 4, 2)
+            mask = torch.zeros((self._boxes.shape[0], height, width), dtype=self.dtype, device=self.device)
+
+        # Boxes coordinates can be outside the image size after transforms. Clamp values to the image size
+        clipped_boxes_xyxy = self.to_tensor("xyxy")
+        clipped_boxes_xyxy[..., ::2].clamp_(0, width)
+        clipped_boxes_xyxy[..., 1::2].clamp_(0, height)
+
+        # Reshape mask to (BxN, H, W) and boxes to (BxN, 4) to iterate over all of them.
+        # Cast boxes coordinates to be integer to use them as indexes. Use round to handle decimal values.
+        for mask_channel, box_xyxy in zip(mask.view(-1, height, width), clipped_boxes_xyxy.view(-1, 4).round().int()):
+            # Mask channel dimensions: (height, width)
+            mask_channel[box_xyxy[1] : box_xyxy[3], box_xyxy[0] : box_xyxy[2]] = 1
+
+        return mask
+
+    def transform_boxes(self, M: torch.Tensor) -> "Boxes":
+        r"""Function that applies a transformation matrix to the boxes.
+
+        Args:
+            M: The transformation matrix to be applied, shape of :math:`(3, 3)` or :math:`(B, 3, 3)`.
+        Returns:
+            The transformed boxes.
+        """
+        if not 2 <= M.ndim <= 3 or M.shape[-2:] != (3, 3):
+            raise ValueError(f"The transformation matrix shape must be (3, 3) or (B, 3, 3). Got {M.shape}.")
+
+        # Due to some torch.jit.script bug (at least <= 1.9), you need to pass all arguments to __init__ when
+        # constructing the class from inside of a method.
+        return Boxes(_transform_boxes(self._boxes, M), False)
```

Comment:
I am wondering should we include a ```in_place``` arg to determine if to return a new bbox or transform locally.

Reply:
Maybe, we could also add `transform_boxes_` like pytorch does for inplace operations. What do you think?